src/api/event-services/work-request.js

```diff
@@ -28,13 +33,17 @@ const handleWorkRequest = async (workRequest, socket) => {
       await validateRequest(workRequest, 'WorkRequest.v1.yaml');
       const { timeout } = workRequest;
       if (Date.parse(timeout) <= Date.now()) {
+        // Annotate current segment as expired.
+        AWSXRay.getSegment().addAnnotation('result', 'error-timeout');
+
         throw new Error(`Work request will not be handled as timeout of ${timeout} is in the past...`);
       }
 
       const workSubmitService = new WorkSubmitService(workRequest);
       await workSubmitService.submitWork();
     } else {
       logger.log('Unexpected error happened while trying to process cached response:', e.message);
+      AWSXRay.getSegment().addError(e);
```

Comment:
Should we create a new logger method to take care of regular console logging + `AWSXRay.getSegment().addError(e)`? Maybe it would make less likely that we forgot about xray-tracing errors...

Reply:
We should add this as a backlog item. I think it's a good idea, but it requires a bit more thinking, e.g. `addError` can take a full `Error` and automatically print the traces. If we just printed the text, we would lose important information.